src/Temporalio.Extensions.Hosting/README.md

```diff
@@ -0,0 +1,93 @@
+# Hosting and Dependency Injection Support
+
+This extension adds support for worker
+[Generic Host](https://learn.microsoft.com/en-us/dotnet/core/extensions/generic-host)s and activity
+[Dependency Injection](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection) to the
+[Temporal .NET SDK](https://github.com/temporalio/sdk-dotnet).
+
+⚠️ UNDER ACTIVE DEVELOPMENT
+
+This SDK is under active development and has not released a stable version yet. APIs may change in incompatible ways
+until the SDK is marked stable.
+
+## Quick Start
+
+Add the `Temporalio.Extensions.Hosting` package from [NuGet](https://www.nuget.org/packages/Temporalio). For example,
+using the `dotnet` CLI:
+
+    dotnet add package Temporalio.Extensions.Hosting --prerelease
+
+To create a worker service, you can do the following:
+
+```csharp
+using Temporalio.Extensions.Hosting;
+
+var builder = Host.CreateApplicationBuilder(args);
+// Add a hosted Temporal worker which returns a builder to add activities and workflows
+builder.Services.
+    AddHostedTemporalWorker(
+        "my-temporal-host:7233",
+        "my-namespace",
+        "my-task-queue).
+    AddScopedActivities<MyActivityClass>().
+    AddWorkflow<MyWorkflow>();
+
+// Run
+var host = builder.Build();
+host.Run();
+```
+
+This creates a hosted Temporal worker which returns a builder. Then `MyActivityClass` is added to the service collection
+as scoped (via
+`TryAddScoped`(https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.extensions.servicecollectiondescriptorextensions.tryaddscoped)) and registered on the worker. Also `MyWorkflow` is registered as a
+workflow on the worker.
+
+This means that for every activity invocation, a new scope is created and the the activity is obtained via the service
+provider. So if it is registered as scoped the activity is created each time but if it registered as singleton it is
+created only once and reused. The activity's constructor can be used to accept injected dependencies.
+
+Workflows are inherently self-contained, deterministic units of work and therefore should never call anything external.
+Therefore, there is no such thing as dependency injection for workflows, their construction and lifetime is managed by
```

Comment:
Is it worth mentioning that to load configuration into workflows, we recommend using local activities?

Reply:
Not here I don't think. If there are general docs where we want to state that it may be worth it.